cpp/CMakeLists.txt

```diff
@@ -689,6 +688,39 @@ following IMPORTED GLOBAL  targets:
     ]=])
 
 
+set(common_code_string
+    [=[
+if(NOT TARGET cudf::Thrust)
+  thrust_create_target(cudf::Thrust FROM_OPTIONS)
+endif()
+
+# nvcc automatically adds the CUDA Toolkit system include paths before any
+# system include paths that CMake adds.
+#
+# CMake implicitly treats all includes on import targets as 'SYSTEM' includes.
```

Comment:
Hmmm, so this is the root cause? Would it be possible to make this configurable in a future CMake version to avoid all the below complexity?

Reply:
Yes the root cause is implicit behavior in CMake, and I am working on upstreaming solutions to CMake for this issue.